### DISCUSSION TITLE
Does GitHub Actions override my compilation options?

### DISCUSSION THREAD

**Discussion by MatthieuHernandez** [community/community#26888] [Actions]
<p>When I added <code>-Wall -Wextra -pedantic -Werror</code> compile options for GCC and <code>/W4 /WX</code> on my project obviously my code didn’t work anymore and I need to fix all the warnings but before to fixed them I pushed just the modification on CMakeList.txt the GitHub Actions of my project passed with GCC and MSVC.<br>
Weird ? I check several time… right branch, right workflow, all seems right.<br>
Does GitHub Actions override my compilation options ?</p>

**Comment by airtower-luna** [comment]
<p>It really shouldn’t, but could you share how the options are set and a workflow log? Maybe there is some odd twist. :slightly_smiling_face:</p>

**Comment by MatthieuHernandez** [comment]
<p>Thanks for your answer. I just discovered that my ‘windows’ run on Microsoft Windows Server 2019 but compile with GNU 8.1.0. How is it possible ? GNU is only Linux compiler, no ?<br>
This is my workflow:</p>
<p><em>unit_tests.yml</em></p>
<pre><code class="lang-auto">name: Unit tests Linux/Windows

on: [push]

jobs:
  unit-tests:
    name: Build and run unit tests
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]

    steps:
    - name: Checkout
      uses: actions/checkout@v2
    - name: CMake
      run: |
        cd ./build
        cmake -G"Unix Makefiles" ./..
    - name: Make
      run: |
        cd ./build
        make
    - name: Execute unit tests
      run: |
        cd ./build
        ./tests/unit_tests/UnitTests.out --gtest_also_run_disabled_tests
</code></pre>
<p>How can I change my workflow to compile with MSVC on Windows and with GCC on Linux ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>GCC is available for many platforms, including Windows. Using it would explain why MSVC compiler options are ignored. :slightly_smiling_face: Or does the Linux build seem to ignore the options, too?</p>
<p>Your workflow doesn’t seem to explicitly configure a specific compiler, so I assume you should check how to configure a preference in CMake.</p>

**Comment by MatthieuHernandez** [comment]
<p>Thanks for information. Do I need to configure a preference in CMake or just configure the compiler in the workflow ? I will check it soon as possible.<br>
Also for compile GCC on windows GitHub must use something like MinGW or Cywin ?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="matthieuhernandez" data-post="5" data-topic="132656">
<div class="title">
<div class="quote-controls"></div>
 matthieuhernandez:</div>
<blockquote>
<p>Do I need to configure a preference in CMake or just configure the compiler in the workflow ?</p>
</blockquote>
</aside>
<p>I’m not sure about CMake specifically, but a common thing on Linux is to set the C compiler via the <code>CC</code> environment variable, so setting that in the CMake run might be worth a try.</p>
<aside class="quote no-group" data-username="matthieuhernandez" data-post="5" data-topic="132656">
<div class="title">
<div class="quote-controls"></div>
 matthieuhernandez:</div>
<blockquote>
<p>Also for compile GCC on windows GitHub must use something like MinGW or Cywin ?</p>
</blockquote>
</aside>
<p>The current version of the <a href="https://github.com/actions/virtual-environments/blob/597569e608ad1cc27fbc3536730ce621323e8b6c/images/win/Windows2019-Readme.md"><code>windows-2019</code> environment description</a> lists <a href="http://mingw-w64.org/doku.php/start">Mingw-w64</a>, so I assume that’s what it is. :wink:</p>
